### PR TITLE
Add `withBuilder` to expose the underlying builder as an extension point

### PR DESCRIPTION
**Pull request by checketts**
I apologize for the slew of PRs. This PR adds a `withBuilder` method that exposes the underlying `MappingBuilder` and `ResponseDefinitionBuilder`

This allows exposing the rest of the WireMock APIs without having to reproduce them in the DSL. That way users can add extension functions that extend the DSL.

Note the mapper code had to be moved into the specifications classes to access the private `builders` variables, but otherwise is largely unchanged.

### CODE REVIEW

File: src/main/kotlin/com/marcinziolo/kotlin/wiremock/RequestSpecification.kt [hunk 1]
```diff
@@ -10,6 +15,93 @@ class RequestSpecification {
...
+
+    internal fun toMappingBuilder(method: Method): MappingBuilder =
+            urlBuilder(method)
+                    .also { headersBuilder(it) }
+                    .also { bodyBuilder(it) }
+                    .also { cookieBuilder(it) }
+                    .also { queryParamsBuilder(it) }
+                    .also { it.atPriority(priority) }
+                    .also { scenarioBuilder(it) }
+                    .also { this.builders.forEach { b -> it.b() } }
```
Comment (checketts) [inline src/main/kotlin/com/marcinziolo/kotlin/wiremock/RequestSpecification.kt:32]: Mapping the builders lambdas to the builder

File: src/main/kotlin/com/marcinziolo/kotlin/wiremock/ResponseSpecification.kt [hunk 2]
```diff
@@ -16,6 +19,33 @@ class ResponseSpecification {
...
+                .withBody(body.trimIndent())
+
+        when (val delay = delay.value) {
+            is FixedDelay -> responseDefinitionBuilder.withFixedDelay(delay.delay)
+            is NormalDistributionDelay ->
+                responseDefinitionBuilder.withLogNormalRandomDelay(delay.median.toDouble(), delay.sigma)
+        }
+        __headers.forEach { (key, values) -> responseDefinitionBuilder.withHeader(key, *values.toTypedArray()) }
+
+        this.builders.forEach { b -> responseDefinitionBuilder.b() }
```
Comment (checketts) [inline src/main/kotlin/com/marcinziolo/kotlin/wiremock/ResponseSpecification.kt:43]: Mapping the response builder lambdas to the builder.

File: src/test/kotlin/com/marcinziolo/kotlin/wiremock/otherPackage/WithBuilderTest.kt [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+import org.hamcrest.Matchers.equalTo
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+
+class WithBuilderTest : AbstractTest() {
+    @Test
+    fun `response with two headers`() {
+        wiremock.get {
+            url equalTo "/hello"
+            withBuilder {
```
Comment (checketts) [inline src/test/kotlin/com/marcinziolo/kotlin/wiremock/otherPackage/WithBuilderTest.kt:17]: Inline usage of `withBuilder`

File: src/test/kotlin/com/marcinziolo/kotlin/wiremock/otherPackage/WithBuilderTest.kt [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+            get("$url/hello")
+        } Then {
+            statusCode(200)
+            extract().headers().getValues("TraceId").also { values ->
+                assertEquals(listOf("abcd123", "efgh456"), values)
+            }
+        }
+    }
+
+    fun ResponseSpecification.json(status: Int = 200) {
```
Comment (checketts) [inline src/test/kotlin/com/marcinziolo/kotlin/wiremock/otherPackage/WithBuilderTest.kt:59]: Extension method using `withBuilder`

File: src/test/kotlin/com/marcinziolo/kotlin/wiremock/otherPackage/WithBuilderTest.kt [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+            withHeader("Content-Type", "application/json")
+        }
+    }
+
+    @Test
+    fun `returns json`() {
+        wiremock.get {
+            url equalTo "/hello"
+        } returns {
+            json()
```
Comment (checketts) [inline src/test/kotlin/com/marcinziolo/kotlin/wiremock/otherPackage/WithBuilderTest.kt:71]: Extension usage

File: src/test/kotlin/com/marcinziolo/kotlin/wiremock/otherPackage/WithBuilderTest.kt [hunk 6]
```diff
@@ -0,0 +1,84 @@
+package com.marcinziolo.kotlin.wiremock.otherPackage
```
Comment (checketts) [inline src/test/kotlin/com/marcinziolo/kotlin/wiremock/otherPackage/WithBuilderTest.kt:1]: Note this test is in an `otherPackage` to illustrate the `import`s that are needed to stuff like `equalTo`, etc.

**Comment by marcinziolo** [comment]
Nice way of providing user original wiremock api :)
Thanks, great idea :)

**Pull request merged**